### DISCUSSION TITLE
Support for KWS-302WF / Tuya WiFi 8 in 1 Power Meter 2P Multi-function AC Energy Meter Color Display APP Control 170-270V/63A Voltage and Current Meter

### DISCUSSION THREAD

**Discussion by pkmaxl** [make-all/tuya-local#2109] [Q&A]
Hello i just got this device from aliexpress. It is working in the tuya app on my mobile phone but i cant get it into home assistant. Does anybody know how to get this working in home assistant? When i add it in home asistant it will be recongnized as climate device or remote control or sprinkler but not as power meter.

Link to the device:
https://de.aliexpress.com/item/1005006228450635.html?spm=a2g0o.detail.0.0.1b03PFCTPFCTDS&mp=1&gatewayAdapt=glo2deu

Thanks for your help in advance
yours
Peter

**Comment by sambhavjain12** [comment]
Hello, I myself just recently got hold of this device and have been tinkering with it. 
I am also having the same issue, I was able to add to Home assistant using Tuya cloud integration, with which I was only able to control turning on and off. I could not get any energy attributes from this device via cloud. 

Commenting here to let you know that it is possible via Cloud but without attributes. Will keep you updated once I manage to get it working with Tuya Local and all the energy attributes

**Comment by Surmeneli61TR** [comment]
Help me :) 
Only on off switch

**Reply by make-all** [reply]
Are you using the latest release? Please file a bug report if it is not detected by the latest release.

**Reply by dangelomda** [reply]
https://github.com/make-all/tuya-local/issues/2856#issuecomment-2621733699

**Comment by make-all** [comment]
Support was added in tuya-local 2024.9.0, after the request #2257 provided the required details for this device.

**Comment by dangelomda** [comment]
For KWS-303WF

https://github.com/make-all/tuya-local/issues/2856#issuecomment-2621733699

**Comment by adesmiroze** [comment]
yo tengo el mismo problema solo me reconoce como un interruptor. que puedo hacer ya registré incluso en la nube y nada. no se como puedo obtner los datos en home assitant.

**Reply by dangelomda** [reply]
https://github.com/make-all/tuya-local/issues/2856#issuecomment-2621733699

**Comment by Franz4596** [comment]
Hello, I have a **KWS-306WF** Switch-Power-Meter from AIiexpress. 
![grafik](https://github.com/user-attachments/assets/664a2090-b3c6-4ed7-a98d-4ff9f43ad605)
 It works well with Tuya/SmartLife-APP and so I integrated it in Home Assistant OS Core 2025.6.3 Tuya local because I assume its like the suported _Gaqqee KWS-302WF energy meter. 
The switch works, but all other entenities (energy, power, voltage, amps) have wrong data.  
Like @sambhavjain12 wrote, the Tuya cloud integration works with on/off-switch and energy display only.
What can I do get good values ?  Thanks for answers.
